airflow/providers/google/cloud/secrets/secret_manager.py

```diff
@@ -89,11 +92,12 @@ def __init__(
         self.variables_prefix = variables_prefix
         self.config_prefix = config_prefix
         self.sep = sep
-        if not self._is_valid_prefix_and_sep():
-            raise AirflowException(
-                "`connections_prefix`, `variables_prefix` and `sep` should "
-                f"follows that pattern {SECRET_ID_PATTERN}"
-            )
+        if connections_prefix is not None:
+            if not self._is_valid_prefix_and_sep():
+                raise AirflowException(
+                    "`connections_prefix`, `variables_prefix` and `sep` should "
+                    f"follows that pattern {SECRET_ID_PATTERN}"
+                )
```

Comment:
This one is a little different from the others, as it checks if the prefix and sep are valid. But the function `_is_valid_prefix_and_sep` only seems to really check the `connections_prefix` with `sep`, and not `variables_prefix` or `config_prefix`. So the added condition reflects this behavior.